devito/data/decomposition.py

```diff
@@ -230,9 +230,17 @@ def convert_index(self, *args, rel=True):
                 elif isinstance(glb_idx, slice):
                     if self.loc_empty:
                         return slice(-1, -3)
-                    glb_idx_min = self.glb_min if glb_idx.start is None else glb_idx.start
-                    glb_idx_max = self.glb_max if glb_idx.stop is None else glb_idx.stop-1
-                    retfunc = lambda a, b: slice(a, b + 1, glb_idx.step)
+                    if glb_idx.step >= 0:
+                        glb_idx_min = self.glb_min if glb_idx.start is None \
+                            else glb_idx.start
+                        glb_idx_max = self.glb_max if glb_idx.stop is None \
+                            else glb_idx.stop-1
+                        retfunc = lambda a, b: slice(a, b + 1, glb_idx.step)
+                    else:
+                        glb_idx_max = self.glb_max if glb_idx.start is None \
```

Comment:
are you sure this is not ``self.glb_min if glb_idx.start is None else glb_idx.start`` ?
ie ``self.glb_min`` instead of ``self.glb_max`` ? I haven't thought about this too much, but it seems off that you pick the glb_max if None *and* the slice start (ie the **min**) otherwise

Reply:
If we're stepping backwards `slice.stop` < `slice.start`: hence the flip. Does that make sense to you?

Reply:
ahhhh! sure sorry my bad